kgtk/cli/visualize-force-graph.py

```diff
@@ -117,266 +275,432 @@ def run(input_file: KGTKFiles,
+              edges.append({'source': row[n1], 'target': row[n2], 'label': row[l2][1:row[l2].find('@')-1], 'width_orig': width_orig})
+          else:
+              edges.append({'source': row[n1], 'target': row[n2], 'label': row[l2], 'width_orig': width_orig})
+
+
+        if edge_width_mapping == 'fixed':
+          for edge in edges:
+            if edge['width_orig'] >= 0:
+              edge['width'] = edge['width_orig']
+            else:
+              edge['width'] = edge_width_default
+        elif edge_width_scale == 'linear':
+          arr = []
+          for edge in edges:
+            if edge['width_orig'] >= 0:
+              arr.append(edge['width_orig'])
+          for edge in edges:
+            if edge['width_orig'] >= 0:
+              edge['width'] = edge_width_minimum + (edge['width_orig'] - min(arr)) * (edge_width_maximum - edge_width_minimum) / (max(arr) - min(arr))
+            else:
+              edge['width'] = edge_width_default
+        elif edge_width_scale == 'log':
+          arr = []
+          for edge in edges:
+            if edge['width_orig'] >= 0:
+              arr.append(edge['width_orig'])
+          for edge in edges:
+            if edge['width_orig'] >= 0:
+              if min(arr) == 0:
+                log_min = 0
+              else:
+                log_min = math.log(min(arr), base)
+
+              if max(arr) == 0:
+                log_max = 0
+              else:
+                log_max = math.log(max(arr), base)
+
+              if edge['width_orig'] == 0:
+                log_cur = 0
+              else:
+                log_cur = math.log(edge['width_orig'], base)
+
+              edge['width'] = edge_width_minimum + (log_cur - log_min) * (edge_width_maximum - edge_width_minimum) / (log_max - log_min)
+            else:
+              edge['width'] = edge_width_default
+
+
 
         d['nodes'] = []
         d['links'] = edges
+        color_set = {}
+        count = 0
+
+        node_color = 0
 
         if 'None' in node_file:
             for ele in nodes:
-                d['nodes'].append({'id': ele})
-        elif '.tsv' in node_file:
-            df = pd.read_csv(node_file, sep = '\t')
-            for i in range(0, len(df)):
-                if 'x' in df.columns:
-                    h = df['color'][i].lstrip('#')
-                    s = 'rgba' + str(tuple(int(h[i:i+2], 16) for i in (0, 2, 4)))
-                    d['nodes'].append({'id': df['node;label'][i][1:df['node;label'][i].find('@')-1], 'color': s, 'fx': float(df['x'][i]), 'fy': float(df['y'][i])})
-                else:
-                    h = df['color'][i].lstrip('#')
-                    s = 'rgba' + str(tuple(int(h[i:i+2], 16) for i in (0, 2, 4)))
-                    d['nodes'].append({'id': df['node;label'][i], 'color': s})
+                d['nodes'].append({'id': ele[0], 'label': ele[1]})
         else:
-            df = pd.read_csv(node_file)
+            if '.tsv' in node_file:
```

Comment:
Use `KGTKReader` for reading node or edge files, it can take care of this file extension logic

Reply:
changed to KGTKReader